integrations/firehydrant/client.py

```diff
@@ -0,0 +1,451 @@
+        self, questions: list[dict[str, Any]]
+    ) -> list[dict[str, Any]]:
+        """Format retrospective questions by renaming title to 'question' and body to 'answer'.
+
+        Args:
+            questions (list[dict[str, Any]]): A list of retrospective question objects.
+
+        Returns:
+            list[dict[str, Any]]: A list of formatted retrospective questions with 'question' and 'answer' keys.
+        """
+
+        retro_responses_list = []
+        for question in questions:
+            retro_responses_list.append(
+                {"question": question.get("title"), "answer": question.get("body")}
+            )
+        return retro_responses_list
+
+    async def compute_service_mean_time_metrics(
+        self, active_incidents_ids: list[Any]
+    ) -> dict[str, Any]:
+        """Computes mean time service details for active incidents.
+
+        Args:
+            active_incidents_ids (list[Any]): A list of active incident IDs.
+
+        Returns:
+            dict[str, Any]: A dictionary containing computed mean time service metrics,
+            including time to acknowledge, time to identify, time to mitigate, and time to resolve.
+
+        Note:
+            This function calculates various incident analytics metrics based on milestone data.
+            The metrics may include Mean Time to Detection (MTTD), Mean Time to Acknowledged (MTTA),
+            Mean Time to Mitigation (MTTM), and Mean Time to Resolution (MTTR).
+        """
+
+        if len(active_incidents_ids) == 0:
+            return {}
+
+        service_metrics = {
+            "time_to_acknowledge": 0,
+            "time_to_identify": 0,
+            "time_to_mitigate": 0,
+            "time_to_resolve": 0,
+        }
+
+        for incident_id in active_incidents_ids:
+            incident_data = await self.get_single_incident(incident_id=incident_id)
+            milestones = incident_data.get("milestones", {})
+            incident_metrics = await self.compute_incident_analytics_metrics(
+                milestones=milestones
+            )
+
+            # Update the cumulative metrics for the service
+            for metric_name, metric_value in incident_metrics.items():
+                service_metrics[metric_name] += metric_value
+
+        # Calculate the average metrics for the service
+        for metric_name in service_metrics:
+            service_metrics[metric_name] //= len(active_incidents_ids)
+
+        return service_metrics
+
+    async def get_single_incident(self, incident_id: str) -> dict[str, Any]:
+        """Retrieve information about a single incident by its unique identifier.
+
+        Args:
+            incident_id (str): The unique identifier of the incident to retrieve.
+
+        Returns:
+            dict[str, Any]: A dictionary containing information about the specified incident.
+
+        """
+        return await self.send_api_request(endpoint=f"incidents/{incident_id}")
```

Comment:
is there a reason u r getting all the incidents one by one?

is it possible to fetch them all and cache it 
or using some kind of caching in this client for requests that might be duplicated